### PR TITLE
feat!: automatically detect contentType if not provided

### LINKED ISSUE #1184: Automatically set content-type based on file type
While uploading a file to a Storage Bucket, the API automatically sets the content-type to `application/octet-stream` instead of setting it to what it should be. e.g. image/png etc.

 **Describe the solution you'd like**
Instead of having to manually set this metadata or it being set to `application/octet-stream`, let the API automatically set this based on the file extension.
 **Describe alternatives you've considered**
Check for each file type extension and then set the metadata manually.

### PR DESCRIPTION
**Pull request by atn**
Fixes #1184  🦕

In a rush, I didn't have time to edit tests.ts file. It would be appreciated if someone could!

### CODE REVIEW

**Comment by austtn** [comment]
@googlebot I signed it!

**Comment by stephenplusplus** [comment]
Thank you! I will take it over :)

File: src/bucket.ts [hunk 1]
```diff
@@ -3438,12 +3438,6 @@ class Bucket extends ServiceObject {
       });
     }
 
-    const contentType = mime.contentType(path.basename(pathString));
```
Comment (stephenplusplus) [inline src/bucket.ts:3441]: Removed as this is identically covered inside of `createWriteStream()` when this code makes its way there.

**Comment by austtn** [comment]
@googlebot I consent.

**Review by jkwlui — APPROVED** [review]

**Pull request merged**